File: leetcode/217-contains-duplicate/ContainsDuplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> trackingSet = new HashSet<>();
        
        for (int num : nums) {
            if (trackingSet.contains(num)) {
                return true;
            }
            trackingSet.add(num);
        }
        
        return false;
    }
}

File: leetcode/217-contains-duplicate/contains-duplicate.js
const containsDuplicate = nums => {
    if (!nums || !Array.isArray(nums) || !nums.length) {
        return false;
    }
    
    nums.sort();
    
    for (let i = 1; i < nums.length; i++) {
        if (nums[i] === nums[i - 1]) {
            return true;
        }
    }
    
    return false;
}
